test/support/api_case.ex

```diff
@@ -0,0 +1,33 @@
+defmodule ElixirGoogleScraperWeb.ApiCase do
+  use ExUnit.CaseTemplate
+
+  using do
+    quote do
+      alias ExOauth2Provider.AccessTokens
+      alias ExOauth2Provider.Applications
+
+      def authenticated_request(user, conn, target, type, params \\ %{}) do
```

Comment:
How about use the `setup` block to return authenticated `conn` instead? Then in the test, we use that `conn` to make a request. 

You can check about setup: block here https://hexdocs.pm/phoenix/testing.html#the-conncase

Reply:
As discussed, I used a helper function to authenticate the `conn`